TAScheduler/UnitTests/tests.py

```diff
@@ -76,71 +77,879 @@ class TestAdminCourseInstrAsgmt(TestCase):  # Kevin
+    def test_OverCap(self):
+        self.taObj.assignTALab(self.labDBList[0])
+        self.taObj.assignTALab(self.labDBList[1])
+        with self.assertRaises(ValueError,
+                               msg="can't assign courses when already reaches the max TA assignments"):
+            self.taObj.assignTALab(self.labDBList[2])
+
+    # 5] Trying to add a non-lab
+    def test_NonLab(self):
+        invalid_inputs = [123, 3.14, True, [1, 2, 3], {'key': 'value'}]  # testing a bunch of different obj types
+
+        for invalid_input in invalid_inputs:
+            with self.subTest(
+                    invalid_input=invalid_input):  # if 1 subtest test runs, it will continue running through loop
+                with self.assertRaises(ValueError, msg="Shouldn't be allowed to assign TA to non-course"):
+                    self.taObj.assignTALab(invalid_input)
+
+    # 6] Assigning lab TA w/ grader status
+    def test_GraderStatus(self):
+        tempUser = User.objects.create(email_address='grader@gmail.com')  # HOPEFULLY DON'T NEED ALL FIELDS?
+        tempTa = TA.objects.create(user=tempUser, max_assignments=2, grader_status=True)  # new TA in db W/ GraderStatus
+        self.taObj = TAObj(tempTa)
+        with self.assertRaises(RuntimeError, msg="TA can't assign to lab when grader"):
+            self.taObj.assignTALab(self.labDBList[0])
 
 
 class TestTAGetTALabAsgmts(TestCase):  # Kiran
-    pass
-
-
-class TestAssignTALec(TestCase):
-    pass
+    taDB = None
+    labDB = None
+    user = None  # for TA
+    taObj = None
+    adminObj = None  # for deleting course
+
+    def setUp(self):
+        self.user = User.objects.create(
+            email_address='TA@example.com',
+            password='TApassword',
+            first_name='TA',
+            last_name='User',
+            home_address='123 TA St',
+            phone_number='1234567890'
+        )
+        self.taDB = TA.objects.create(user=self.user, max_assignments=1)  # max 1 assignment!
+        self.taObj = TAObj(self.taDB)
+        tempUserForAdmin = User(email_address="admin@example.com")  # HOPEFULLY OK W/O FIELDS?
+        tempAdmin = Administrator(user=tempUserForAdmin)
+        self.adminObj = AdminObj(tempAdmin)
+
+        # Course
+        self.courseDB = Course.objects.create(
+            course_id=101,
+            semester='Fall 2023',
+            name='Introduction to Testing',
+            description='A course about writing tests in Django.',
+            num_of_sections=3,
+            modality='Online',
+            credits=4)
+        # Section
+        self.sectionDB = Section.objects.create(
+            section_id=100 + 1,
+            course=self.courseDB,
+            location="location" + str(1),
+            meeting_time="mt" + str(1))
+        # Lab
+        Lab.objects.create(
+            section_id=self.sectionDB,
+            ta=None)
+
+    # 1] 1 lab assignment
+    def test_1Assignment(self):
+        self.taObj.assignTALab(self.labDB)
```

Comment:
Again I would not use the methods that could full of errors in testing of different methods. It just seems like bad practice